### PR TITLE
Add notification functions

### PR DESCRIPTION
**Pull request by dshanske**
This adds functions to enhance notification emails and messages sent using two different plugins(Pushover and Pushbullet) to support Semantic Linkbacks to properly display the information.

It tries to keep to the original style as much as possible.

It also adds a debug function that will send an email or message from Bulk Actions in Comments to manually test this functionality.

Also switches hook from comment_text, which is only when echoed to the screen, to get_comment_text.

### CODE REVIEW

**Comment by dshanske** [comment]
I originally wrote the first version of this code during Indiewebcamp Cambridge in I think 2015.

**Comment by pfefferle** [comment]
This is very specific and only a few people will profit from this feature, so I would prefer a more "plugin" like style, to keep the code as simple as possible. I added something similar in the ZenPress theme: https://github.com/pfefferle/ZenPress/blob/master/functions.php#L446 https://github.com/pfefferle/ZenPress/tree/master/integrations

**Comment by dshanske** [comment]
To be honest, the fact that the emails sent for webmentions do not have context isn't a local issue. The bigger annoyance is that WordPress requires you to rewrite the entire email text in order to make this work.

I'm thinking of rewriting https://core.trac.wordpress.org/ticket/33735, which was not my best work, to address same, but, Comments are not something they seem to want to give much attention to lately.

**Comment by dshanske** [comment]
I'm going to split this into its own file though.

**Comment by dshanske** [comment]
@pfefferle Split the file out.

File: includes/class-linkbacks-notifications.php [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+		if ( ! $type ) {
+			return $subject;
+		}
+		$title    = get_the_title( $comment->comment_post_ID );
+		$blogname = wp_specialchars_decode( get_option( 'blogname' ), ENT_QUOTES );
+		/* translators: 1: blog name, 2: Semantic Linkbacks type, 3: post title */
+		return sprintf( __( '[%1$s] %2$s: %3$s', 'semantic-linkbacks' ), $blogname, $type, $title );
+	}
+
+	/**
```
Comment (pfefferle) [inline includes/class-linkbacks-notifications.php]: indents :)

File: includes/class-linkbacks-notifications.php [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+		$notify_message .= self::moderate_text( $comment );
+		return $notify_message;
+	}
+
+	/**
+	 * Filter a notification text
+	 * @param string $notify_messsage
+	 * @param int|WP_Comment $comment
+	 * @return string $notify_basic
+	 **/
```
Comment (pfefferle) [inline includes/class-linkbacks-notifications.php]: only one `*`

**Comment by pfefferle** [comment]
nice! that looks much cleaner!

**Pull request merged**